lib/common/utils/device_utils.dart

```diff
@@ -0,0 +1,18 @@
+import 'dart:io';
+import 'package:android_id/android_id.dart';
+import 'package:device_info_plus/device_info_plus.dart';
+
+/// Device related utility functions.
+class DeviceUtils {
+
+  /// Retrieves device ID for Android or iOS.
+  static Future<String?> getId() async {
+    if (Platform.isIOS) {
```

Comment:
Fungovalo by toto? Možná mi to přijde více elegantnější, ale netuším, zda-li to bude rozumně fungovat.
```suggestion
   switch (Platform.operatingSystem) {
      case 'ios':
        return (await DeviceInfoPlugin().iosInfo).identifierForVendor;
      case 'android':
        return await AndroidId().getId();
      default:
        return null;
    }
```

Reply:
Yeap, it should definitely work 👍